foldersync-desktop: init at 2.8.5

LouisJalouzot opened the pull request:
Add FolderSync Desktop, a file synchronization and backup application supporting 20+ cloud storage providers.

- Homepage: https://foldersync.io/
- License: Proprietary (unfree)

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LouisJalouzot committed:
maintainers: add LouisJalouzot

LouisJalouzot commented on pkgs/by-name/fo/foldersync-desktop/package.nix:
```diff
@@ -0,0 +1,212 @@
+    passthru.updateScript = ./update.sh;
+  };
+
+  darwin = stdenv.mkDerivation {
```
Unfortunately I don't have access to any macOS machine. It would be nice if someone could give it a try.

gaelj reviewed (commented):
For Linux this looks (to me) / builds / runs OK. But I can't 'blindly' approve the untested darwin part.

@GaetanLepage I believe you have a darwin machine - would you mind checking this out please ?

GaetanLepage commented on pkgs/by-name/fo/foldersync-desktop/package.nix:
```diff
@@ -0,0 +1,206 @@
+  };
+in
+(if stdenv.hostPlatform.isDarwin then darwin else linux)
+// {
```
I'm not sure to understand this part.

GaetanLepage commented on pkgs/by-name/fo/foldersync-desktop/package.nix:
```diff
@@ -0,0 +1,206 @@
+      url = "https://github.com/tacitdynamics/foldersync-desktop-production/releases/download/${version}/foldersync-desktop-${version}-mac-aarch64.zip";
+      hash = "sha256-uEopiDsJVKxws3nVdxEzR78AVuYqsHyKqAK669wronU=";
+    };
+  };
```
```suggestion
    sources = {
      x86_64-linux = {
        arch = "linux-amd64";
        hash = "...";
      };
      aarch64-linux = {
        arch = "linux-aarch64";
        hash = "...";
      };
      x86_64-darwin = {
        arch = "mac-amd64";
        hash = "...";
      };
      aarch64-darwin = {
        arch = "mac-aarch64";
        hash = "...";
      };
    };
```

LouisJalouzot commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 475273`
Commit: [`0803909e7d465b7c36342a47f593af3f23d0b3d6`](https://github.com/NixOS/nixpkgs/commit/0803909e7d465b7c36342a47f593af3f23d0b3d6) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/0803909e7d465b7c36342a47f593af3f23d0b3d6..pull/475273/head))
Merge: [`bb4c52f76e74693cc301836591e925070090a80d`](https://github.com/NixOS/nixpkgs/commit/bb4c52f76e74693cc301836591e925070090a80d)

Logs: https://github.com/LouisJalouzot/nixpkgs-review-gha/actions/runs/21098095081


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foldersync-desktop</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foldersync-desktop</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foldersync-desktop</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>foldersync-desktop</li>
  </ul>
</details>

LouisJalouzot commented:
Thank you @GaetanLepage for testing the Darwin builds and suggesting the `finalAttrs` pattern refactoring!

I made the following changes:
- Added `unzip` to Darwin `nativeBuildInputs` to fix the unpacking issue
- Refactored to use the `finalAttrs` pattern with cleaner sources structure
- Each derivation now contains its own meta block

I could check that the build succeeds on all four platforms thanks to [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha) (see results above).
Code is ready for re-review when convenient.

Thanks again @gaelj and @GaetanLepage for your time!

GaetanLepage reviewed (commented):
I think that you should not split this in a `linux` and `darwin` items.
You could have a single derivation with some attributes that are conditional.

If you keep it this way, then move the common items in a `let in` block.

gaelj commented on pkgs/by-name/fo/foldersync-desktop/package.nix:
```diff
@@ -0,0 +1,215 @@
+
+  passthru.updateScript = ./update.sh;
+
+  linux = stdenv.mkDerivation {
```
Don't forget to use `finalAttrs.version`

gaelj reviewed (changes requested):
Please update to current version

gaelj reviewed (approved):
Builds and starts fine on x86-64_linux

SuperSandro2000 commented on pkgs/by-name/fo/foldersync-desktop/update.sh:
```diff
@@ -0,0 +1,35 @@
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+latestVersion=$(curl -sSfL https://api.github.com/repos/tacitdynamics/foldersync-desktop-production/releases/latest | jq -r '.tag_name')
```
```suggestion
latestVersion=$(curl -sSfL ${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} https://api.github.com/repos/tacitdynamics/foldersync-desktop-production/releases/latest | jq -r '.tag_name')
```

SuperSandro2000 commented on pkgs/by-name/fo/foldersync-desktop/package.nix:
```diff
@@ -0,0 +1,215 @@
+      appPath=$(find "$out/Applications" -name "*.app" -maxdepth 1 | head -1)
+      if [ -n "$appPath" ]; then
+        makeWrapper "$appPath/Contents/MacOS/"* "$out/bin/foldersync-desktop" || true
+      fi
```
Please use nix conditions with isLinux or isDarwin.

SuperSandro2000 commented on pkgs/by-name/fo/foldersync-desktop/package.nix:
```diff
@@ -0,0 +1,215 @@
+      ext = if stdenv.hostPlatform.isDarwin then "zip" else "tar.gz";
+    in
+    fetchurl {
+      url = "https://github.com/tacitdynamics/foldersync-desktop-production/releases/download/${version}/${pname}-${version}-${arch}.${ext}";
```
```suggestion
      url = "https://github.com/tacitdynamics/foldersync-desktop-production/releases/download/${version}/foldersync-desktop-${version}-${arch}.${ext}";
```

LouisJalouzot committed:
foldersync-desktop: init at 2.8.5
